apps/web/app/(app)/stats/NewsletterModal.tsx

```diff
@@ -57,8 +76,15 @@ export function NewsletterModal(props: {
                   href={newsletter.lastUnsubscribeLink || undefined}
                   target="_blank"
                   rel="noreferrer"
+                  onClick={onUnsubscribe}
                 >
-                  Unsubscribe
+                  {!unsubscribeLoading && <span>Unsubscribe</span>}
+                  {!!unsubscribeLoading && (
+                    <div className="flex cursor-not-allowed items-center opacity-50">
+                      <ButtonLoader />
+                      <span>Unsubscribing...</span>
+                    </div>
+                  )}
```

Comment:
Could use a ternary to make this cleaner. ie: `unsubscribeLoading ? Unsub : Loading...`

If you're trying to add disabled state, then you could add disabled={unsubscribeLoading} to the button itself.

Reply:
Updated it now! Used a ternary here instead and added the disabled state to the button.